aikau/src/main/resources/alfresco/services/PreferenceService.js

```diff
@@ -98,11 +88,36 @@ define(["dojo/_base/declare",
       localPreferences: null,
       
       /**
+       * Converts a dot-notation property address into a JavaScript object literal. This is done so that
+       * preferences can be set on the object generated from and address.
+       * 
+       * @instance
+       * @param {string} str The dot-notation property to convert into an object
+       * @param {object} value The value to set at the dot-notation property address
+       * @return {object} The object generated from the dot-notation property.
+       */
+      dotNotationToObject: function alfresco_services_PreferenceService__dotNotationToObject(str, value) {
+         var object = {}, obj = object;
+         if (typeof str === "string")
+         {
+            var properties = str.split("."), property, i, ii;
+            for (i = 0, ii = properties.length - 1; i < ii; i++)
+            {
+               property = properties[i];
+               obj[property] = {};
+               obj = obj[property];
+            }
+            obj[properties[i]] = value !== undefined ? value : null;
+         }
+         return object;
+      },
+
```

Comment:
Because I took the code verbatim from alfresco.js out of the Share project. Also, I'm not sure that getObject does what we want here... we want to create an object from a string, not inspect an existing object with a string